components/camel-spring/src/main/java/org/apache/camel/spring/SpringCamelContext.java

```diff
@@ -306,4 +308,24 @@ public String toString() {
         return sb.toString();
     }
 
+    @Override
+    public int getPhase() {
+        // the context is started by invoking start method which
+        // happens either on ContextRefreshedEvent or explicitly
+        // invoking the method, for instance CamelContextFactoryBean
+        // is using that to start the context, _so_ here we want to
+        // have maximum priority as the getPhase() will be used only
+        // for stopping, in order to be used for starting we would
+        // need to implement SmartLifecycle which we cannot
+        // (explained in comment in the onApplicationEvent method)
+        // we use LOWEST_PRECEDENCE here as this is taken into account
+        // only when stopping and then in reversed order
+        return LOWEST_PRECEDENCE;
+    }
+
+    @Override
+    public boolean isRunning() {
+        return isStarted();
```

Comment:
I think it's fine like this, this would affect Spring lifecycle calling start/stop. So for start we are not using lifecycle (as the code stands now), and for stop we could have just as easily returned `true` and let ServiceSupport handle multiple stop invocations (if any). Perhaps a more correct way would be to return `!isStopping() && !isStopped()`.